hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -286,7 +286,11 @@ private boolean syncSchema(String tableName, boolean tableExists, boolean useRea
           config.getBooleanOrDefault(HIVE_SUPPORT_TIMESTAMP_TYPE));
       if (!schemaDiff.isEmpty()) {
         LOG.info("Schema difference found for " + tableName);
-        syncClient.updateTableSchema(tableName, schema);
+        if (!schemaDiff.getAddColumnTypes().isEmpty() || !schemaDiff.getUpdateColumnTypes().isEmpty()) {
```

Comment:
we should always keep schema up to date. when later data written with old schema, https://hudi.apache.org/docs/configurations#hoodiedatasourcewritereconcileschema this config is to adapt the data into the new schema. so i don't think we should skip update schema